test/e2e/predictor/test_sklearn.py

```diff
@@ -249,3 +251,112 @@ def test_sklearn_v2_grpc():
+                                       name=service_name, namespace=KSERVE_TEST_NAMESPACE),
+                                   spec=V1beta1InferenceServiceSpec(predictor=predictor))
+
+    kserve_client = KServeClient(
+        config_file=os.environ.get("KUBECONFIG", "~/.kube/config"))
+    kserve_client.create(isvc)
+    kserve_client.wait_isvc_ready(
+        service_name, namespace=KSERVE_TEST_NAMESPACE)
+
+    response = predict(service_name, "./data/sklearn_mixed_v2.json", protocol_version="v2")
+    assert response["outputs"][0]["data"] == [12.202832815138274]
+
+    kserve_client.delete(service_name, KSERVE_TEST_NAMESPACE)
+
+
+@pytest.mark.slow
+def test_sklearn_v2_mixed_grpc():
+    service_name = "isvc-sklearn-v2-mixed-grpc"
+    model_name = "sklearn"
+    predictor = V1beta1PredictorSpec(
+        min_replicas=1,
+        model=V1beta1ModelSpec(
+            model_format=V1beta1ModelFormat(
+                name="sklearn",
+            ),
+            runtime="kserve-sklearnserver",
+            storage_uri="gs://kfserving-examples/models/sklearn/1.0/mixedtype",
+            resources=V1ResourceRequirements(
+                requests={"cpu": "50m", "memory": "128Mi"},
+                limits={"cpu": "100m", "memory": "512Mi"},
+            ),
+            ports=[
+                V1ContainerPort(
+                    container_port=8081,
+                    name="h2c",
+                    protocol="TCP"
+                )],
+            args=["--model_name", model_name, "--protocol", "grpc-v2"]
+        )
+    )
+
+    isvc = V1beta1InferenceService(api_version=constants.KSERVE_V1BETA1,
+                                   kind=constants.KSERVE_KIND,
+                                   metadata=client.V1ObjectMeta(
+                                       name=service_name, namespace=KSERVE_TEST_NAMESPACE),
+                                   spec=V1beta1InferenceServiceSpec(predictor=predictor))
+
+    kserve_client = KServeClient(
+        config_file=os.environ.get("KUBECONFIG", "~/.kube/config"))
+    kserve_client.create(isvc)
+    kserve_client.wait_isvc_ready(
+        service_name, namespace=KSERVE_TEST_NAMESPACE)
+
+    json_file = open("./data/sklearn_mixed.json")
+    data = json.load(json_file)
+    payload = []
+    for key, val in data.items():
+        if isinstance(val, int):
+            payload.append(
+                {
+                    "name": key,
+                    "shape": [1],
+                    "datatype": "INT32",
+                    "contents": {
+                        "int_contents": [val]
+                    }
+                }
+            )
+        elif isinstance(val, str):
+            payload.append(
+                {
+                    "name": key,
+                    "shape": [1],
+                    "datatype": "BYTES",
+                    "contents": {
+                        "bytes_contents": [bytes(val, 'utf-8')]
+                    }
+                }
+            )
```

Comment:
We probably need an util function to convert from pandas data framework to open inference protocol